app/src/components/avatar_event_bus.rs

```diff
@@ -0,0 +1,55 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use yew_agent::{Agent, AgentLink, Context, HandlerId};
+
+use super::avatar::AvatarData;
```

Comment:
Can you make that an absolute import? `crate::infra::avatar...`
Although the type AvatarData should probably be defined here rather than in the component.